workmail-message-flow-state-machine/README.md

```diff
@@ -4,43 +4,34 @@ This application enables you to use AWS Step Functions to orchestrate multiple A
-
-Depending on your use case, you may choose to raise the value of WAIT_TIME_FOR_EXECUTION if you know the state machine will 
-return in a predictable amount of time. Note that the Lambda function timeout is set to 60 seconds and you can only
-increase it to 15 minutes, which is shorter than the 240 minutes that WorkMail will retry the function. 
-Also, long running Lambda functions incur more costs. So, it is recommended that you keep WAIT_TIME_FOR_EXECUTION set to a low value.
+If the Step Function state machine does not finish execution before `WaitTimeForExecution` or the `MachineStateForOutput` result is not available by that time, then the state machine's executionArn is saved to a Dynamo DB table with a TTL of 240 minutes (maximum value of `Rule timeout`) and the Lambda function will return an error signaling to WorkMail to retry again later. 
+
+WorkMail will keep re-invoking the function until it gets a result or `Rule timeout` is reached. Upon each subsequent invocation lambda function will look up the executionArn of the state machine from the Dynamo DB table and check for result. If no result is returned back to WorkMail before 240 minutes then the rule's default action will take effect.
+
+Depending on your use case, you may choose to raise the value of `WaitTimeForExecution` if you know the state machine will return in a predictable amount of time of in order of seconds since long running Lambda functions incur costs.
 
 ## Modifying your Step Functions state machine
 
-Once you have finished the setup, send a test email message in to your WorkMail mailbox. 
-From the Step Functions console you will see a successful execution. 
+Once you have finished the setup, send a test email message in to your WorkMail mailbox. From the Step Functions console you will see a successful execution. 
 In the execution, expand ExecutionSucceeded to see the output from the state machine's execution; this output was used by the Lambda function to return to the WorkMail Message Flow API.
 
-You can configure the state machine to return different output based on the logic you want to define. 
-Since the input to the state machine is the same as the input that was sent to the WorkMail Message Flow Lambda, you can pass this input into additional states within your state machine that execute other WorkMail functions; which you can subsequently use the results from other functions to return back via the execution output.
+You can configure the state machine to return different output based on the logic you want to define. Since the input to the state machine is the same as the input that was sent to the WorkMail Message Flow Lambda, you can pass this input into additional states within your state machine that execute other WorkMail functions; which you can subsequently use the results from other functions to return back via the execution output.
```

Comment:
...to return **a** different output...

... **from** which you can ~~subsequently~~ use the results **of** other functions...